lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -220,6 +220,7 @@ export class UserAgentApplication {
         this.telemetryManager = this.getTelemetryManagerFromConfig(this.config.system.telemetry, this.clientId);
 
         AuthorityFactory.setKnownAuthorities(this.config.auth.validateAuthority, this.config.auth.knownAuthorities);
+        AuthorityFactory.parseAuthorityMetadata(this.config.auth.authority, this.config.auth.authorityMetadata);
```

Comment:
Should we write a helper for all authority initiation code? Looks like we are doing one too many things when we initiate authority instance.

Reply:
I can add this idea to the cloud instance PR after this PR gets merged as I'll be adding a few more steps on initiation in that PR.